applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,368 +57,300 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
+     * updated.
+     * </p>
+     *
+     * @param configurationSetting The configuration setting to create or update.
+     * @return ConfigurationSetting that was created or updated.
+     * @throws com.azure.common.http.rest.RestException If the {@link ConfigurationSetting#etag()} was specified, is not
+     *                                               {@link ConfigurationClient#ETAG_ANY}, and the current configuration
+     *                                               value's etag does not match.
      */
-    public Mono<RestResponse<ConfigurationSetting>> getKeyValue(String key, KeyValueFilter filter) {
-        if (key == null || key.isEmpty()) {
-            throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
-        }
-        if (filter != null) {
-            return service.getKeyValue(baseUri.toString(), key, filter.label(), filter.fields(),
-                    filter.acceptDateTime(), filter.ifMatch(), filter.ifNoneMatch());
-        }
-        return service.getKeyValue(baseUri.toString(), key, null, null, null, null, null);
+    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting) {
+        Validator.validate(configurationSetting);
+        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters()
+                .withValue(configurationSetting.value())
+                .withContentType(configurationSetting.contentType())
+                .withTags(configurationSetting.tags());
+
+        return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, getETagValue(configurationSetting.etag()), null);
     }
 
+
     /**
-     * Deletes the ConfigurationSetting.
-     * @param key keyValue to delete
-     * @return the deleted ConfigurationSetting or none if didn't exist.
+     * Updates an existing configuration value in the service. The setting must already exist.
+     *
+     * <p>
+     * The label value for the ConfigurationSetting is optional. If not specified, the
+     * {@link ConfigurationSetting#NULL_LABEL} is used.
+     * </p>
+     *
+     * If the {@link ConfigurationSetting#etag()} is specified, the configuration value is only updated if it matches.
+     *
+     * @param configurationSetting The key, value, and optional label to set.
+     * @return ConfigurationSetting that was updated.
+     * @throws com.azure.common.http.rest.RestException When a ConfigurationSetting with the same key and label does not
+     *                                               exists or the configuration value is locked.
      */
-    public Mono<RestResponse<ConfigurationSetting>> deleteKeyValue(String key) {
-        return deleteKeyValue(key, null, null);
+    public Mono<RestResponse<ConfigurationSetting>> updateKeyValue(ConfigurationSetting configurationSetting) {
```

Comment:
Consider adding the overload methods for add/set/update where the user only specifies key and optional label. 
See https://microsoft.sharepoint.com/teams/AzureDeveloperExperience/_layouts/OneNote.aspx?id=%2Fteams%2FAzureDeveloperExperience%2FShared%20Documents%2FService%20-%20Azconfig%2FAzure%20SDK%20-%20Service%20-%20Azconfig&wd=target%28Azconfig.one%7C50856656-5D71-2640-B240-11758DB2FB6E%2FStandarization%20across%20languages%7C79180E7A-F7B4-4221-8398-D440467E3782%2F%29

Reply:
Unfortunately, there isn't such a thing as a "default" method parameter. Would having
`add(string key, string label)` be fine? Or should I have two more overloads for `add(key)` and `add(key, label)`

Reply:
If it is that complicated you could leave it to only take the full object and not overload methods.
Once users start using it, we can see what their feedback is and see if we need to implement those methods